.github/workflows/sphinx-docs.yml

```diff
@@ -0,0 +1,64 @@
+name: docs
+on:
+  push:
+    branches: [main]
+  pull_request:
+    paths:
+      - 'docs/**'
```

Comment:
**question:** as we use autoapi to build html from python sources, don't we want to rebuild the website on any change impacting the code? (i.e. not only docs directly)